server.py

```diff
@@ -58,23 +66,24 @@ def contest(code,number):
 
 @app.get("/contest/<code>")
 def contest(code):
-    for contest in contests:
-        if(contest.code == code):
+    for contest_code, contest in contests.items():
+        if(contest_code == code):
             description = contest.description
             questions = contest.questions
-    return template("contest.html", code=code, description=description, questions=questions)
-
-@app.get("/question/<number>")
-def question(number):
-    statement = questions[number].statement
-    return template("index.html", question_number=number, question=statement)
-
+            start_time = contest.start_time
+            end_time = contest.end_time
+            if contest.start_time > datetime.datetime.now():
+                return "The contest had not started yet."
+    return template(
+        "contest.html", code=code, description=description, questions=questions,
```

Comment:
wouldn't it be easier to just pass in `contest=contest` and let the HTML use `contest.code`? Then while reading the html you know exactly what code it is that is being rendered there.

Reply:
code is not part of contest as it is key, so i let `code` stay and passed `contest` for others